### PR TITLE
Implementation of the cleanup script and test deployment

### PR DESCRIPTION
**Pull request by koala7659**
**Cleanup script**

1. Following annotations are applied on all existing servicemappings:
```
io.javaoperatorsdk/primary-name: connectivity-proxy
io.javaoperatorsdk/primary-namespace: kyma-system
```
2. It removes all resources from previous installation of Connectivity Proxy

**Test deployment**
- Deployments with PR container for testing
- Deployments with init container for testing
- RBACs for cluster_admin proviledges

**Related issue**
https://github.com/kyma-project/kyma/issues/18482

### CODE REVIEW

File: images/cleaner/cleaner.sh [hunk 1]
```diff
@@ -1,5 +1,75 @@
...
+kubectl delete clusterrole connectivity-proxy-restart-watcher --ignore-not-found
+kubectl delete serviceaccount -n kyma-system connectivity-proxy-restart-watcher --ignore-not-found
+
+kubectl delete clusterrolebinding connectivity-proxy-service-mappings --ignore-not-found
+kubectl delete clusterrole connectivity-proxy-service-mappings --ignore-not-found
+kubectl delete serviceaccount -n kyma-system connectivity-proxy-sm-operator --ignore-not-found
+
+kubectl delete mutatingwebhookconfiguration connectivity-proxy-mutating-webhook-configuration --ignore-not-found
+
+echo "Removing all Connectivity Proxy ConfigMaps"
```
Comment (akgalwas) [inline images/cleaner/cleaner.sh]: PriorityClass `connectivity-proxy-priority-class` is not removed.

File: hack/test-deployment/deployment-single.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apiVersion: apps/v1
```
Comment (akgalwas) [inline hack/test-deployment/deployment-single.yaml:1]: What is the purpose of this file?

File: hack/test-deployment/rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+apiVersion: v1
```
Comment (akgalwas) [inline hack/test-deployment/rbac.yaml]: `cleaner-sa` seems to not be used anywhere. Is it still needed?

File: hack/test-deployment/deployment-single.yaml (hunk 2 above)
Comment (koala7659) [reply hack/test-deployment/deployment-single.yaml:1]: The purpose of this file is testing

File: images/cleaner/cleaner.sh [hunk 4]
```diff
@@ -1,5 +1,78 @@
...
+kubectl delete deployment -n kyma-system connectivity-proxy-restart-watcher --ignore-not-found
+kubectl delete deployment -n kyma-system connectivity-proxy-sm-operator --ignore-not-found
+
+echo "Removing all Connectivity Proxy services"
+
+kubectl delete service -n kyma-system connectivity-proxy --ignore-not-found
+kubectl delete service -n kyma-system connectivity-proxy-smv --ignore-not-found
+kubectl delete service -n kyma-system connectivity-proxy-tunnel --ignore-not-found
+kubectl delete service -n kyma-system connectivity-proxy-tunnel-0 --ignore-not-found
+kubectl delete service -n kyma-system connectivity-proxy-tunnel-healthcheck --ignore-not-found
```
Comment (akgalwas) [inline images/cleaner/cleaner.sh:41]: `connectivity-proxy-tunnel` is deleted twice. Its not a problem actually, but I will remove it either way

File: images/cleaner/cleaner.sh [hunk 5]
```diff
@@ -1,5 +1,78 @@
 #!/bin/sh
 
+set -e
+
 echo "Running Connectivity Proxy Cleanup script"
-kubectl get pods -n kyma-system
-exit 0
\ No newline at end of file
+
```
Comment (akgalwas) [inline images/cleaner/cleaner.sh:6]: `ValidatingWebhookConfiguration/webhook-secret` is not deleted

**Review by akgalwas — APPROVED** [review]

**Pull request merged**